src/main/java/com/startsteps/ecommerceapi/user/model/Role.java

```diff
@@ -5,19 +5,22 @@
 import lombok.NoArgsConstructor;
 import lombok.Setter;
 
-@Entity
+import java.util.Collection;
+import java.util.Set;
+
 @Table(name = "roles")
 @NoArgsConstructor
 @Getter
 @Setter
 public class Role {
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
-    private Integer id;
+    private Long id;
     @Enumerated
     @Column(length = 20)
     private UserRoles roleName;
     String description;
     String permission;
-    Long userId;
+    @ManyToMany(mappedBy = "roles")
+    Set<User> users;
```

Comment:
why are you mapping Users to the Role entity?

Reply:
if I got your question correctly, the user is the owner of relationships (therefore we use a join in user entity), however, one user can have many roles (customer and seller, for ex). which also should help with the implementation of role-based authentication (user can access the information based on the role). please correct me if I am wrong.

Reply:
You already have information about the user roles inside your User entity class. So i think this is not needed here.